src/renderer/cursor_renderer/mod.rs

```diff
@@ -341,6 +347,54 @@ impl CursorRenderer {
 
         canvas.restore();
 
+        // draw ime if possible under current cursor
+        if let Some(preedit) = preedit {
+            let factor = grid_renderer.font_dimensions.width as f32;
+            let y_adjustment = y_adjustment as f32 + factor / 5.0;
+
+            let off = if let Some(off) = preedit.cursor_offset() {
+                off as f32 * factor
+            } else {
+                0f32
+            };
+
+            let mut start = self.destination.x;
+            let mut end = self.destination.x + off;
+            if end + factor > w {
```

Comment:
I think ideally we should use the neovim grid window width here, not the global window width, otherwise it will render outside floating window and splits for example.

This als means that the code better belongs in `rendered_window`, where the width and tranaparency needed is available.